qualtran/bloqs/qubitization_walk_operator.py

```diff
@@ -103,8 +105,12 @@ def decompose_from_registers(
         reflect_reg = {reg.name: quregs[reg.name] for reg in self.reflect.signature}
         reflect_op = self.reflect.on_registers(**reflect_reg)
         for _ in range(self.power):
-            yield select_op
-            yield reflect_op
+            if self.is_adjoint:
+                yield cirq.inverse(reflect_op)
+                yield cirq.inverse(select_op)
+            else:
+                yield select_op
+                yield reflect_op
```

Comment:
@tanujkhattar there was no bloq adjoint for `QubitizationWalkOperator`, so I fixed it manually. Ideally this should be fixed properly when refactoring `QubitizationWalkOperator`.

Reply:
why does the default fallback implementation of `Bloq.adjoint()` not work here?

Reply:
```py
from qualtran.bloqs.qsp.generalized_qsp import GeneralizedQSP
from qualtran.bloqs.qubitization_walk_operator_test import get_walk_operator_for_1d_ising_model

W = get_walk_operator_for_1d_ising_model(2, 1e-4)
G = GeneralizedQSP.from_qsp_polynomial(W, (0.5, 0, 0.5), negative_power=1)
G.decompose_bloq()
```
produces
```
ValueError: Cannot control channel with non-unitary operators: Adjoint(subbloq=QubitizationWalkOperator(select=SelectPauliLCU(selection_bitsize=2, target_bitsize=2, select_unitaries=(cirq.DensePauliString('ZZ', coefficient=(2+0j)), cirq.DensePauliString('XI', coefficient=(-1+0j)), cirq.DensePauliString('IX', coefficient=(-1+0j))), control_val=None), prepare=StatePreparationAliasSampling(selection_registers=(Register(name='selection', dtype=BoundedQUInt(bitsize=2, iteration_length=3), shape=(), side=<Side.THRU: 3>),), alt=array([0, 0, 0]), keep=array([   0, 3072, 3072]), mu=12), control_val=None, power=1))
```

Because QubitizationWalkOperator still uses cirq style API to provide the decompositions.

Reply:
this has been reverted now, the default implementation works after overriding `Adjoint.controlled`